guides/source/active_record_basics.md

```diff
@@ -220,134 +263,192 @@ The application will have to use the [`id_value`][] alias attribute to access th
 Rails will throw an error during migrations such as:
-`you can't redefine the primary key column 'id' on 'my_products'.`
+`you can't redefine the primary key column 'id' on 'my_books'.`
 `To define a custom primary key, pass { id: false } to create_table.`
 
 CRUD: Reading and Writing Data
 ------------------------------
 
 CRUD is an acronym for the four verbs we use to operate on data: **C**reate,
 **R**ead, **U**pdate and **D**elete. Active Record automatically creates methods
-to allow an application to read and manipulate data stored within its tables.
+to allow you to read and manipulate data stored in your application's database tables.
+
+Active Record makes it seamless to perform CRUD operations by using these high-level methods that abstract away database access details. Note that all of these convenient methods result in SQL statement(s) that are executed against the underlying database.
+
+The examples below show a few of the CRUD methods as well as the resulting SQL statements.
 
 ### Create
 
 Active Record objects can be created from a hash, a block, or have their
 attributes manually set after creation. The `new` method will return a new
 object while `create` will return the object and save it to the database.
 
-For example, given a model `User` with attributes of `name` and `occupation`,
+For example, given a `Book` model with attributes of `title` and `author`,
 the `create` method call will create and save a new record into the database:
 
 ```ruby
-user = User.create(name: "David", occupation: "Code Artist")
+book = Book.create(title: "The Lord of the Rings", author: "J.R.R. Tolkien")
 ```
 
-Using the `new` method, an object can be instantiated without being saved:
+While the `new` method will instantiate an object without saving it to the database:
 
 ```ruby
-user = User.new
-user.name = "David"
-user.occupation = "Code Artist"
-```
+book = Book.new
+book.title = "The Hobbit"
+book.author = "J.R.R. Tolkien"
+
+# The above `book` is not yet saved to the database
+
+book.save
 
-A call to `user.save` will commit the record to the database.
+# Now the `book` record is committed to the database.
+```
 
 Finally, if a block is provided, both `create` and `new` will yield the new
 object to that block for initialization, while only `create` will persist
 the resulting object to the database:
 
 ```ruby
-user = User.new do |u|
-  u.name = "David"
-  u.occupation = "Code Artist"
+book = Book.new do |b|
+  b.title = "Metaprogramming Ruby 2"
+  b.author = "Paolo Perrotta"
 end
+
+book.save
+```
+
+The resulting SQL statement from both `book.save` and `Book.create` look something like this:
+
+```sql
+INSERT INTO "books" ("title", "author", "created_at", "updated_at") VALUES (?, ?, ?, ?) RETURNING "id"  [["title", "Metaprogramming Ruby 2"], ["author", "Paolo Perrotta"], ["created_at", "2024-02-22 20:01:18.469952"], ["updated_at", "2024-02-22 20:01:18.469952"]]
 ```
```

Comment:
It just occurred to me that we could maybe add a note here to mention how `created_at` and `updated_at` got set automatically, since that was kinda mentioned in the migration block... do you think that would add to the example?

Same for `updated_at` in the update subsection/example.

Not sure we'd need to mention about the id. Also feel free to skip/ignore it, if it doesn't seem like a good addition.

Reply:
Yeah I think that would. I added a little comment in the sql block for both `INSERT` and `UPDATE`.